File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    tangerine_map = {}
    for t in tangerine:
        if t in tangerine_map:
            tangerine_map[t] += 1
        else:
            tangerine_map[t] = 1
    
    sorted_tangerine_map = sorted(tangerine_map.items(), key = lambda x: x[1], reverse = True)
    answer = 0
    
    for size, cnt in sorted_tangerine_map:
        k -= cnt
        answer += 1
        
        if k <= 0:
            break
            
    return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    let total = 0;
    const tangerineMap = new Array(Math.max(...tangerine)).fill(0)
    tangerine.forEach(el => tangerineMap[el - 1]++)
    tangerineMap.sort((a, b) => b - a)
    
    for (let i = 0; i < tangerineMap.length; i++) {
        total += tangerineMap[i]
        answer++
        
        if (total >= k) break
    }
    
    return answer;
}